From 1f3f8f988516394330541fc35eb58e0ddddbc827 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 2 Oct 2023 01:16:49 -0500
Subject: [PATCH 3/5] loader: CMake related static hacks

Adds additional hacks for CMake's side. Does not (yet) go for parity
with shinchiro's patch yet.

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 loader/CMakeLists.txt | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 001adb9cd..cd9ed71c3 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -318,12 +318,22 @@ if(WIN32)
         set(RC_FILE_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/loader.rc)
     endif()
 
+
+    set(LOADER_UNKNOWN_CHAIN_LIBRARY $<$<TARGET_EXISTS:loader-unknown-chain>:$<TARGET_OBJECTS:loader-unknown-chain>>)
+
+    if (BUILD_STATIC_LOADER)
+        set(VULKAN_LIBRARY_TYPE STATIC)
+    else()
+        set(VULKAN_LIBRARY_TYPE SHARED)
+    endif()
+
     add_library(vulkan
-                SHARED
+                ${VULKAN_LIBRARY_TYPE}
                 ${NORMAL_LOADER_SRCS}
                 ${LOADER_UNKNOWN_CHAIN_LIBRARY}
                 ${CMAKE_CURRENT_SOURCE_DIR}/${API_TYPE}-1.def
-                ${RC_FILE_LOCATION})
+                ${RC_FILE_LOCATION}
+                ${OPT_LOADER_SRCS})
 
     target_link_libraries(vulkan PRIVATE loader_specific_options)
 
@@ -452,7 +462,7 @@ if (LOADER_USE_UNSAFE_FILE_SEARCH)
 endif()
 
 # common attributes of the vulkan library
-target_link_libraries(vulkan PRIVATE loader_specific_options)
+target_link_libraries(vulkan PRIVATE $<BUILD_INTERFACE:loader_specific_options>)
 
 target_link_libraries(vulkan PRIVATE Vulkan::Headers)
 add_library(Vulkan::Loader ALIAS vulkan)
-- 
2.45.0

